test/powershell/Language/Scripting/DeserializedTypeConversion.Tests.ps1

```diff
@@ -0,0 +1,283 @@
+        $t6 = [PSClass2]::new("PSClassName2", 80, "1..5")
+    }
+    
+
+    Context 'Type conversion and parameter binding of deserilized Type case 1: type definition contains public fields' { 
+        BeforeAll {
+            $s = [System.Management.Automation.PSSerializer]::Serialize($t1)
+            $dst1 = [System.Management.Automation.PSSerializer]::Deserialize($s)
+        }
+
+        It 'Type casts should succeed.' {
+            { $tc1 = [test1]$dst1 }| Should not Throw
+        }
+
+        It 'Parameter bindings should succeed.' {
+            { test-1 $dst1 } | Should Not Throw
+        }
+    }
+
+    Context 'Type conversion and parameter binding of deserilized Type case 2: type definition contains public properties' {
+        BeforeAll {
+            $s = [System.Management.Automation.PSSerializer]::Serialize($t2)
+            $dst2 = [System.Management.Automation.PSSerializer]::Deserialize($s) 
+        }
+        It 'Type casts should succeed.' {                   
+            { $tc2 = [test2]$dst2 } | Should Not Throw
+        }
+
+        It 'Parameter bindings should succeed.' {
+            { test-2 $dst2 } | Should Not Throw
+        }
+    }    
+
+    Context 'Type conversion and parameter binding of deserilized Type case 1: type definition contains 2 public properties and 1 read only property' {
+        BeforeAll {
+            $s = [System.Management.Automation.PSSerializer]::Serialize($t3)
+            $dst3 = [System.Management.Automation.PSSerializer]::Deserialize($s)
+        }      
+
+        It 'Type casts should fail.' {
+        
+            try
+            {
+                $tc3 = [test3]$dst3
+                Throw "Exception expected, execution should not have reached here"
+            }
+            catch {
+                $_.FullyQualifiedErrorId | Should Be 'InvalidCastConstructorException'
+            }
+        }
+
+        It 'Parameter bindings should fail.' {
+            try
+            {
+                test-3 $dst3
+                Throw "Exception expected, execution should not have reached here"
+            }
+            catch {
+                $_.FullyQualifiedErrorId | Should Be 'ParameterArgumentTransformationError,test-3'
+            }
+        }
+    }
+
+    Context 'Type conversion and parameter binding of deserilized Type case 1: type definition contains 2 public properties' {        
+        BeforeAll {
+            $s = [System.Management.Automation.PSSerializer]::Serialize($t4)
+            $dst4 = [System.Management.Automation.PSSerializer]::Deserialize($s)
+        }
+                
+        It 'Type casts should succeed.' {                   
+            { $tc4 = [test4]$dst4 } | Should Not Throw
+        }
+
+        It 'Parameter bindings should succeed.' {
+            { test-4 $dst4 } | Should Not Throw
+        }
+    }
+
+    Context 'Type conversion and parameter binding of deserilized Powershell class with default constructor' {
```

Comment:
fixed the this typo in the whole file

---
In reply to: [78839104](https://github.com/Powershell/Powershell/pull/2118#discussion_r78839104) [](ancestors = 78839104)